core/unix/signal.c

```diff
@@ -359,6 +359,11 @@ block_all_signals_except(kernel_sigset_t *oset, int num_signals,
         kernel_sigdelset(&set, va_arg(ap, int));
     }
     va_end(ap);
+    /* We never block SIGSEGV or SIGBUS: we need them for various safe reads and to
+     * properly report crashes.
+     */
+    kernel_sigdelset(&set, SIGSEGV);
```

Comment:
Why did you not add the signals to the call of block_all_signals_except() and instead baked them into the function? Ok, if you had a good reason for it, otherwise I would add it to the except list of the function call, since that's what it was meant for.

Reply:
I'm looking at this function as blocking all signals that can *safely* be blocked.  This way we avoid shooting ourselves in the foot the next time this function is used if we forget to add these two.  They should really *never* be blocked for us.  If you look at this function as doing whatever the caller says, then I agree the caller should add them: but I believe that every single caller should add these 2, so we'll have duplication with the risk of messing up somewhere where we forget.  Maybe rename the function?

Reply:
I'm thinking s/block_all_signals_except/block_all_noncrash_signals_except/.

Reply:
My interpretation is influenced by this being exported out of the signal module to all of unix/: that makes it seem like not a low-level utility for the signal module to manipulate signals irrespective of DR, but a DR-centric operation.